### PR TITLE
Adds fuzz targets for DTLS

### PR DESCRIPTION
**Pull request by fwh-dc**
I'd like to add fuzzing of DTLS.

So as a start here are the source files for building such a fuzzer on client and server side.

I've copied the implementation from TLS and modified it to fit DTLS. I'm not sure if it is the best way.

Then I've run the fuzzers: `./fuzz/dtlsclient ../fuzz/corpora/client` and `./fuzz/dtlsserver ../fuzz/corpora/server` and it has generated some corpora that is suitable for DTLS. I was thinking that I could add those to the corpora repo, but I don't know if you have some requirements for adding such initial corpora?

Please advise how to add a job for ci. Does the ci automatically add new corpora to the corpora repo?

##### Checklist

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by fwh-dc** [comment]
@kroeckx I guess you might be able to advise?

**Comment by fwh-dc** [comment]
@t8m I think this could go directly to master? It is not DTLS 1.3 specific. But you decide of course.

**Comment by t8m** [comment]
> @t8m I think this could go directly to master? It is not DTLS 1.3 specific. But you decide of course.

Ah, you're right.

**Comment by kroeckx** [comment]
This is missing the test/recipes/99-test_fuzz_*.t files

File: fuzz/build.info [hunk 1]
```diff
@@ -73,6 +77,14 @@ IF[{- !$disabled{"fuzz-afl"} || !$disabled{"fuzz-libfuzzer"} -}]
   INCLUDE[ct]=../include {- $ex_inc -}
   DEPEND[ct]=../libcrypto {- $ex_lib -}
 
+  SOURCE[dtlsclient]=dtlsclient.c driver.c fuzz_rand.c
+  INCLUDE[dtlsclient]=../include {- $ex_inc -}
+  DEPEND[dtlsclient]=../libcrypto ../libssl {- $ex_lib -}
+
+  SOURCE[dtlsserver]=dtlsserver.c driver.c fuzz_rand.c
+  INCLUDE[dtlsserver]=../include {- $ex_inc -}
+  DEPEND[dtlsserver]=../libcrypto ../libssl {- $ex_lib -}
```
Comment (kroeckx) [inline fuzz/build.info:86]: You need to create a dtlsclient-test and dtlsserver-test target too below.

**Comment by fwh-dc** [comment]
@kroeckx Thanks for having a look. I fixed the issues you found. I'll look at the corpus generation next.

**Review by kroeckx — APPROVED** [review]

**Comment by fwh-dc** [comment]
I'll have a look what's causing the failures.

**Review by kroeckx — APPROVED** [review]

**Comment by kroeckx** [comment]
The CI failures don't seem to be relevant.

**Comment by fwh-dc** [comment]
@t8m can someone from OTC have a look at this one?

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Squashed and merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**